modules/auxiliary/gather/memcached_extractor.rb

```diff
@@ -81,6 +81,20 @@ def enumerate_slab_ids
     slab_ids.uniq! || []
   end
 
+  def enumerate_keys_lru
+    keys = []
+    sock.send("lru_crawler metadump all\r\n", 0)
+    loop do
+      data = sock.recv(4096)
+      break if !data || data.length == 0
+      matches = data.scan(/^key=(?<key>.*) exp=/)
+      keys = keys + matches.flatten! if matches
```

Comment:
Hey, thanks for the PR. I've just tested this against a fresh memcached instance running in docker.

In one tab I ran:

```
docker run --rm -p 11211:11211 memcached
```

And in a msfconsole tab, I ran:

```
use gather/memcached_extractor
set verbose true
set RHOSTS 127.0.0.1
run
```

Which gave the following result:

```
> run

[*] 127.0.0.1:11211       - Connecting to memcached server...
[+] 127.0.0.1:11211       - Connected to memcached version 1.6.6
[*] 127.0.0.1:11211       - Using lru_crawler to enumerate keys
[*] 127.0.0.1:11211       - Error: 127.0.0.1: TypeError no implicit conversion of nil into Array
[*] 127.0.0.1:11211       - Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

It looks like if there's no keys present, memcached immediately returns `END` and causes issues with `keys + matches.flatten!`, as `matches.flatten!` on an empty array returns `nil`:

```
$ irb
2.6.6 :001 > [].flatten!
 => nil 
```

It would be great to fix this edge case - let me know if I can do anything to help! :+1: